### PR TITLE
Feature/issue 2030 coredata remove http error handler global

### LINKED ISSUE #2030: Data: refactor to remove httpErrorHandler global variable
Refactor the above-identified service to remove the above-identified global variable (which is defined in the service's init.go).

This work has already been completed in the system management agent.  Use it as a reference.  See https://github.com/edgexfoundry/edgex-go/blob/master/internal/system/agent/router.go and https://github.com/edgexfoundry/edgex-go/blob/master/internal/system/agent/init.go.

You will need to update route creation to explicitly pass the individual required dependencies to the controller functions.  This will bubble down through the implementation; that is, you'll need to pass those dependencies down the call chain to where they're currently referencing the global variable.  Do not pass the dependency injection container into any controller functions.

You can also create any long-lived dependencies in the service's bootstrap and use the dependency injection container to hold them.  This was done in the system management agent; look there for an example.

Add me as the sole reviewer of your PR.

Note this is Geneva work and should not be merged into master prior to Fuji branching.  As there are a series of these issues for this service (each refactoring away a global variable), assign all of this service's related issues to you and use the branch created for this PR as the starting point for your next PR to avoid extensive conflicts. 

Depends on https://github.com/edgexfoundry/edgex-go/issues/2029

### PR DESCRIPTION
**Pull request by AnthonyMBonafide**
Fix #2030 

Remove the `httpErrorHandler` global variable in favor of using the DI container to pass an instance to the necessary functions.

Update the loadRoutes function to pull the `httpErrorHandler` out of the DI container and pass it to the REST handlers which require it.

### CODE REVIEW

File: internal/core/data/container/http.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
+/********************************************************************************
```
Comment (michaelestrin) [inline internal/core/data/container/http.go]: Use the existing implementation in /internal/pkg/container/errorHandler.go

File: internal/core/data/init.go [hunk 2]
```diff
@@ -95,7 +94,11 @@ func BootstrapHandler(ctx context.Context, wg *sync.WaitGroup, startupTimer star
 		},
 		dataContainer.EventsChannelName: func(get di.Get) interface{} {
 			return chEvents
-		}})
+		},
+		dataContainer.HttpErrorHandlerName: func(get di.Get) interface{} {
+			return httpErrorHandler
```
Comment (michaelestrin) [inline internal/core/data/init.go]: Why not eliminate the httpErrorHandler variable?

`return errorconcept.NewErrorHandler(loggingClient)`

File: internal/core/data/router.go [hunk 3]
```diff
@@ -109,19 +111,35 @@ func LoadRestRoutes(dic *di.Container) *mux.Router {
 	e.HandleFunc(
 		"/"+SCRUB,
 		func(w http.ResponseWriter, r *http.Request) {
-			scrubHandler(w, r, bootstrapContainer.LoggingClientFrom(dic.Get), container.DBClientFrom(dic.Get))
+			scrubHandler(
+				w,
+				r,
+				bootstrapContainer.LoggingClientFrom(dic.Get), container.DBClientFrom(dic.Get),
```
Comment (michaelestrin) [inline internal/core/data/router.go]: One parameter per line.

Comment here; occurs elsewhere.

File: internal/core/data/router.go [hunk 4]
```diff
@@ -109,19 +111,35 @@ func LoadRestRoutes(dic *di.Container) *mux.Router {
...
 	e.HandleFunc(
 		"/"+SCRUBALL,
 		func(w http.ResponseWriter, r *http.Request) {
-			scrubAllHandler(w, r, bootstrapContainer.LoggingClientFrom(dic.Get), container.DBClientFrom(dic.Get))
+			scrubAllHandler(
+				w,
+				r,
+				bootstrapContainer.LoggingClientFrom(dic.Get), container.DBClientFrom(dic.Get),
+				dataContainer.HttpErrorHandlerFrom(dic.Get))
+
```
Comment (michaelestrin) [inline internal/core/data/router.go]: Empty line should be removed (https://github.com/edgexfoundry/edgex-go/pull/2241)

Commented here; occurs elsewhere.

File: internal/core/data/router_test.go [hunk 5]
```diff
@@ -179,10 +179,9 @@ func TestRestValueDescriptorsUsageHandler(t *testing.T) {
 			Configuration = &ConfigurationStruct{Service: bootstrapConfig.ServiceInfo{MaxResultCount: 10}}
 			defer func() { Configuration = &ConfigurationStruct{} }()
 
-			httpErrorHandler = errorconcept.NewErrorHandler(logger.NewMockClient())
 			Configuration.Service = tt.config
 			rr := httptest.NewRecorder()
-			restValueDescriptorsUsageHandler(rr, tt.request, logger.NewMockClient(), tt.dbMock)
+			restValueDescriptorsUsageHandler(rr, tt.request, logger.NewMockClient(), tt.dbMock, errorconcept.NewErrorHandler(logger.NewMockClient()))
```
Comment (michaelestrin) [inline internal/core/data/router_test.go]: Use a single logger mock.

**Comment by AnthonyMBonafide** [comment]
Sorry, forgot to address the PR comments from other PR.

**Comment by AnthonyMBonafide** [comment]
Addressed the code review comments for this PR and for #2241

**Comment by AnthonyMBonafide** [comment]
Recheck

**Comment by michaelestrin** [comment]
recheck

**Comment by michaelestrin** [comment]
Build appears stuck.  @lranjbar

**Comment by jamesrgregg** [comment]
recheck

**Comment by MightyNerdEric** [comment]
recheck

**Pull request merged**